### ISSUE TITLE
QUESTION: Where to find backups/How to store backups outside container

### ISSUE THREAD

**Issue by flow4it** [jokay/docker-ccu-historian#221]
Hello,

first this is not a bug but a question. 

I am using QNAP QTS 5.0 with Container Station 2.6.1 whitch is a Docker system.
At the "store" of container station I found the docker image for historian, build by jokay.

I tried many path variants to store the automatic backup but there is no error from historian or docker. At the command line every day it is simply shown 
`2022-08-27 00:59:58|INFO |Creating backup of database to file HistorianBackup_2022-08-27.zip`

But nowhere on the whole storage I could find this or the files.

So the first question is, where could I find the "successfully" created file? Or is the message from historian not true?

Second question is: How to store the automatic backups outside the docker system. Just on a simple share. I know, that is not in the scope of historian itself but maybe I could find tips here.

I cannot provide docker-compose.yaml and neither the docker run command.
Please tell me how to get them from my environment.

So therefore I would need more specific information how to get to this information.

Actually I think its all default how I gets out of the image, because I did not change anything but the ccu-historian.config whitch is attached.

- OS: QNAP QTS 5.0

- Docker version:
`Client:
 Version:           20.10.17-qnap1
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        0474f29
 Built:             Tue Jun 14 10:41:04 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true
Server: Docker Engine - Community
 Engine:
  Version:          20.10.17-qnap1
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.11
  Git commit:       571a584
  Built:            Tue Jun 14 03:04:25 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.6.6
  GitCommit:        10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
 runc:
  Version:          1.1.2
  GitCommit:        v1.1.2-0-ga916309f
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0`

I was told docker is that much easy, but it seems to bit a little bit difficult just to find a single file.

So I need more help to get more information more like "step by step" manuals.

But I want to thank you @jokay. The container and the historian is working fine and fast, so you did a good job.
If the backup is running it is perfect :)

Best regards,
flow4it

[ccu-historian.zip](https://github.com/jokay/docker-ccu-historian/files/9447932/ccu-historian.zip)

**Comment by jokay** [comment]
I really wonder how you got the backup even to run with the default config, this is why I asked for more information :wink:

The supported environment variables can be found uner [Configuration](https://github.com/jokay/docker-ccu-historian#configuration) any additional config param(s) (e.g. `database.backup`) needs to be added manually in the `ccu-historian.config`

> Additional config settings should be made by changing the config file ccu-historian.config within the docker container.
> It is easier to export the config folder out of the docker container and edit the file there.

As stated on the [documentation](https://github.com/mdzio/ccu-historian/wiki#automatisierte-erstellung-von-backups-der-datenbank) of CCU-Historian itself, you need to specify `database.backup` in order to get the backup work.

> Es können automatisiert Backups der Datenbank erstellt werden. Diese Funktionalität wird mit der Option database.backup konfiguriert. Durch diese Option wird ein Dateipfad mit bestimmten Platzhaltern zur Ablage der erstellten Backup-Dateien angegeben. Relative Dateipfade beziehen sich auf das Arbeitsverzeichnis vom CCU-Historian. Dies ist normalerweise das Installationsverzeichnis.

I do not reinvent the wheel with my container, it only supports the official config params of the application.

If you have a look inside the container the application can be found under the path `/opt/ccu-historian`.

We have mounts for `/database` and `/opt/ccu-historian/config` as mentionned under [volumes](https://github.com/jokay/docker-ccu-historian#volumes).

You could just add another mount yourself for the backup which can be a local mount on your NAS or a remote mount using e.g. NFS by specifying `database.backup='/database/backup/db_%Y-%M-%D-%h'` in `ccu-historian.config`.

Docker is easy, once you understand how it works :wink:

So long story short ...

1) Run the container at least once with the minimum config params (`CONFIG_CCU_TYPE`, `CONFIG_CCU_IP`, `CONFIG_HOST_IP`) and mounts (`/database`, `/opt/ccu-historian/config`)
2) Optional: Add another external mount (e.g. `/database/backup`) by using NFS if you want your backups externally
3) Add the config param `database.backup` in `ccu-historian.config`
4) Restart the container

**Comment by jokay** [comment]
May be the first part of this guide for Synology may be helpful as well, see [here]/https://www.cyberdoc-net.de/2020/10/ccu-historian-daten-mit-grafana-visualisieren/).

**Comment by jokay** [comment]
I only saw the attached `ccu-historian.config` now, try to change ...

```diff
- database.backup='HistorianBackup_%Y-%M-%D.zip'
+ database.backup='/database/HistorianBackup_%Y-%M-%D.zip'
```

**Comment by jokay** [comment]
@flow4it could you setup what you needed?

**Comment by flow4it** [comment]
> @flow4it could you setup what you needed?

I am sorry @jokay but I need more time to get into that container technique. Do you know qnap container station. Frist problem is the environment variables aren't changeable after creating the container. So it could be made by ssh commands but this is difficult as well.
Well anyway I am thinking about setting up a virtual machine with Linux and let it run there. That will make everything so much easier.
Could you recommend a Linux distributon for historian with less requirements?

**Comment by flow4it** [comment]
Sorry closing was not my plan

**Comment by PaulMerk** [comment]
@flow4it
i can change the environment variables, when I stop the container and restart after the changes are saved.

**Comment by jokay** [comment]
The environment variables are only used on the first start of the container to create the `ccu-historian.config`.

You need to create a mount for `/opt/ccu-historian/config` to be able to edit the created `ccu-historian.config` otherwise it won't be persistent. This is how Docker works.

**Comment by flow4it** [comment]
Now I got it running. I had to recreate the container and set it from scratch. Within the wizard I had to set a volume from host in the shares settings.
With /database/backup you mentioned it found the shared path within the container.

The biggest problem for me was to clean a running installation for the backup. But well sometimes there is no other we - as it looks like.

Thanks a lot,
flow4it
